AzAutomation Swagger for job & runbook

najams opened the pull request:
- Swagger for job
- Fixed spec for runbook 
- Added examples for runbook
- Ran validation tools

najams committed:
Added swagger spec for job

najams committed:
fixed unintentional update to
softwareUpdateConfigurationCollectionItemProperties

najams committed:
Examples

najams committed:
Fixed job schema & examples

najams committed:
Fixed description and schema

najams committed:
Merge https://github.com/Azure/azure-rest-api-specs

najams committed:
Fixed runbook and test job and their examples.

najams committed:
Merge https://github.com/Azure/azure-rest-api-specs

najams committed:
Added x-ms-mutability to job and testJob

najams committed:
Merge https://github.com/Azure/azure-rest-api-specs

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-go

vrdmr commented:
@najams - Could you please take a look at the error that the Python build faced - https://github.com/Azure/azure-rest-api-specs/pull/2466#issuecomment-364590497

najams committed:
Fixed build failures

najams committed:
Merge https://github.com/Azure/azure-rest-api-specs

vrdmr commented:
Hi @ravbhatnagar, @sergey-shandar - Could you please take a look at this PR. Thanks.

ravbhatnagar commented on specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/job.json:
```diff
@@ -0,0 +1,613 @@
+            "$ref": "./definitions.json#/parameters/clientRequestId"
+          }
+        ],
+        "responses": {
```
what does this return? does it not have a defined schema.

najams replied:
It returns the output from the script as generated by Powershell. For example, a script with only statement as { get-date } will generate the following output: Saturday, February 17, 2018 10:09:38 PM

whereas, the script like { "hello"; get-date; "bye"; } will generate the output: 
hello

Saturday, February 17, 2018 10:11:15 PM
bye

ravbhatnagar commented on specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/job.json:
```diff
@@ -0,0 +1,613 @@
+            "$ref": "./definitions.json#/parameters/clientRequestId"
+          }
+        ],
+        "responses": {
```
schema?

najams replied:
This will be a string which represents the script used to run the job. As above, there is no schema as it is a script.

ravbhatnagar commented on specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json:
```diff
@@ -12,7 +12,9 @@
   "produces": [
-    "application/json"
+    "application/json",
+    "text/plain",
```
Are you really returning these content types from the ARM control plane? text/plain and text/powershell. Please note that only json is supported.

najams replied:
Yes, these are for the scripts and output as commented above. These are for read-only properties. These APIs are part of ARM since day 1 and signed off by ARM team like Charles, Anders, etc. I only fixed the swagger.

ravbhatnagar commented on specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json:
```diff
@@ -2541,6 +2532,37 @@
+      "description": "The Resource definition.",
+      "x-ms-azure-resource": true
+    },
+    "ProvisioningStateProperty": {
```
enum.

ravbhatnagar commented on specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/runbook.json:
```diff
@@ -48,7 +48,13 @@
           }
         },
+        "produces": [
+          "text/powershell"
```
content type is  an issue

najams committed:
Merge https://github.com/Azure/azure-rest-api-specs

najams committed:
Addressed review comments: 1. make properties read-only 2. Define enum for provisioningState

najams committed:
Fixed examples

najams commented:
Hi @ravbhatnagar, I have addressed your feedback. Please review and let me know if you have more questions. Overall, these are existing GA APIs out for several years. Thanks.

ravbhatnagar commented:
Signing off since these are existing APIs. ARM content type validation needs to be enabled going forward to disallow such behavior.
